src/authn/ccloudStateHandling.ts

```diff
@@ -52,24 +52,49 @@ export class AuthPromptTracker {
  * Called whenever sidecar pushes an update to the ccloud connection via websocket event to us.
  */
 export async function reactToCCloudAuthState(connection: Connection): Promise<void> {
+  const authStatus: CCloudStatus | undefined = connection.status.ccloud;
+  if (!authStatus) {
+    logger.warn("no ccloud status found in connection, skipping auth state handling");
+    return;
+  }
+
+  const status = authStatus as CCloudStatus;
   logger.debug("received update to CCloud connection", {
-    status: connection.status.authentication.status,
-    expiration: connection.status.authentication.requires_authentication_at,
-    errors: connection.status.authentication.errors,
+    state: status.state,
+    expiration: status.requires_authentication_at,
+    errors: status.errors,
   });
 
-  const authStatus: Status = connection.status.authentication.status;
-  observabilityContext.ccloudAuthLastSeenStatus = authStatus;
+  observabilityContext.ccloudAuthLastSeenState = status.state;
+
+  const rm = getResourceManager();
+  const lastAuthState: ConnectedState | undefined = await rm.getCCloudState();
+  await getResourceManager().setCCloudState(authStatus.state);
 
-  await getResourceManager().setCCloudAuthStatus(authStatus);
-  if (authStatus === "INVALID_TOKEN") {
+  const authState: ConnectedState = status.state;
+  // check if the connected state transitioned from SUCCESS to NONE/FAILED
+  const sessionJustExpired: boolean =
+    lastAuthState === ConnectedState.Success &&
+    (authState === ConnectedState.None || authState === ConnectedState.Failed);
+  if (sessionJustExpired) {
+    logger.error(
+      `CCloud session expired: transitioned from ${ConnectedState.Success} to ${authState}`,
+    );
+    // either we got a reauth date, or we're reacting to an expiration that already happened
+    await handleExpiredAuth(
+      new Date(status.requires_authentication_at ?? Date.now()).toLocaleString(),
+    );
```

Comment:
Fix this:
<img width="475" alt="image" src="https://github.com/user-attachments/assets/f159ade5-6029-48a1-be49-95a114c49ca3" />